translation FAQ (#1071)
ropensci/roweb3: content/blog/2025-06-13-translation-faq/index.md

Our [multilingual publishing](/multilingual-publishing) ongoing efforts are crucial work for our [mission](/about).
When sharing about this project at [conferences](/talks-papers/) and [elsewhere](/commcalls/nov2023-multilingual/), we start an important conversation with people less used to such endeavours or to our particular [processes](https://translationguide.ropensci.org/).
In this post, we will summarize frequently asked questions, and our best answers to them (even better formulated than in the heat of a post-talk Q&A session :grin:).

## Will translations make people “lazy” in that they won’t learn English?

First of all, not learning English doesn’t mean being “lazy”. 
Working on one’s project with the tools at one’s disposal is already work enough. 
Besides, many people who do not speak English speak several languages.
Now, people might still choose to learn English later, for instance to take part in some conversations in English, but it won’t have been an issue for them to have first learnt how to, say, develop software using French.

> Do we want people to learn how to use some tools or do we want them to be English speaking? 

Reviewer: Is it "speaking" or "speakers" ?

> In our case, we want people to be able to use and contribute to open-source tools without first having to become English speaking,

Reviewer: Is it "speaking" or "speakers" ?

because we are aware of the "manifold costs of being a non-native English speaker in science"[^amano].

[^amano]: Amano T, Ramírez-Castañeda V, Berdejo-Espinola V, Borokini I, Chowdhury S, Golivets M, et al. (2023) The manifold costs of being a non-native English speaker in science. PLoS Biol 21(7): e3002184. https://doi.org/10.1371/journal.pbio.3002184

### Related: Is it that hard to read in English?

Yes, especially as it adds to the "manifold costs of being a non-native English speaker in science"[^amano] (we really like this paper :wink:).

> 

Reviewer: Some times I also get the comment (more than question) that people should learn English anyway because is the lingua franca of science and programming.  So, perhaps it can be another Related one?  

Here we can mention what was already mentioned and also add the cost on time and money of learning a human language.  Do you think it will add to the discussion? or it is so similar to the other ones?

Author: Yes! Want to add these points?

> 

Reviewer: ```suggestion
There are several studies around the topic of learning to code in English base programming languages when you are not a English speaker. 

Gou and other[^gou] found that non-native English speakers faced barriers with reading instructional materials, technical communication, reading and writing code, and simultaneously learning English and programming. These difficulties are rooted in the need to develop English understanding at the same time that they are expected to learn about the specific topic, in our case how to develop research open source high quality software.

Other results[^tshukudu] suggest that learning to program in English alone, when you are not a English speaker, isn't necessarily difficult for everyone, but including students' native language can make the learning experience clearer, more supportive and more enjoyable. 

If you are interested in learning more about this topic we suggest you follow [Felienne Hermans](https://www.felienne.com), who specializes in programming language design and developed a multilingual programming language for teaching. 

[^gou]: Philip J. Guo. 2018. Non-Native English Speakers Learning Computer Programming: Barriers, Desires, and Design Opportunities. In Proceedings of the 2018 CHI Conference on Human Factors in Computing Systems (CHI '18). Association for Computing Machinery, New York, NY, USA, Paper 396, 1–14. https://doi.org/10.1145/3173574.3173970

[^tshukudu]: Ethel Tshukudu, Emma Dodoo, Felienne Hermans, and Monkgogi Mudongo. 2024. Bilingual Programming: A Study of Student Attitudes and Experiences in the African context. In Proceedings of the 24th Koli Calling International Conference on Computing Education Research (Koli Calling '24). Association for Computing Machinery, New York, NY, USA, Article 42, 1–11. https://doi.org/10.1145/3699538.3699561
```

If you are interested in learning more about this topic we suggest you follow [Felienne Hermans](https://www.felienne.com), who specializes in programming language design and developed a multilingual programming language for teaching. 

[^gou]: Philip J. Guo. 2018. Non-Native English Speakers Learning Computer Programming: Barriers, Desires, and Design Opportunities. In Proceedings of the 2018 CHI Conference on Human Factors in Computing Systems (CHI '18). Association for Computing Machinery, New York, NY, USA, Paper 396, 1–14. https://doi.org/10.1145/3173574.3173970

[^tshukudu]: Ethel Tshukudu, Emma Dodoo, Felienne Hermans, and Monkgogi Mudongo. 2024. Bilingual Programming: A Study of Student Attitudes and Experiences in the African context. In Proceedings of the 24th Koli Calling International Conference on Computing Education Research (Koli Calling '24). Association for Computing Machinery, New York, NY, USA, Article 42, 1–11. https://doi.org/10.1145/3699538.3699561

> ## Why can’t we rely on automatic translation only?

Reviewer: ```suggestion

## Related 2: People Should Learn English

Some people claim that you have to learn English if you want to work in programming and science because English is the lingua franca of these disciplines. 

Even when this is possible and many people have learned English, there are audiences for whom this is less feasible, for example learners, casual or end-user programmers. They might not have the time, energy and money to learn programming and also English[^swidan].

[^swidan]: Alaaeddin Swidan and Felienne Hermans. 2023. A Framework for the Localization of Programming Languages. In Proceedings of the 2023 ACM SIGPLAN International Symposium on SPLASH-E (SPLASH-E 2023). Association for Computing Machinery, New York, NY, USA, 13–25. https://doi.org/10.1145/3622780.3623645

Let’s do some numbers: [The Common European Framework of Reference for Languages](https://www.languagecert.org/en/guided-learning-hours) has tabulate the guided learning hours typically required to reach each language proficiency level. These are hours spent studying with a teacher or structured learning. 

So for example, if you start as a beginner (A1) and want to become intermediate (B1) you will need approximately **400 guided hours**.  If you can dedicate **5 hours a week**, it will take **1.5 years**  and a minimum budget of **USD $6,000** to learn English (to give you an idea, for a person with minimum gauge in Argentina this cost is iqual to two year of salary). 

## Why can’t we rely on automatic translation only?
```

[^swidan]: Alaaeddin Swidan and Felienne Hermans. 2023. A Framework for the Localization of Programming Languages. In Proceedings of the 2023 ACM SIGPLAN International Symposium on SPLASH-E (SPLASH-E 2023). Association for Computing Machinery, New York, NY, USA, 13–25. https://doi.org/10.1145/3622780.3623645

Let’s do some numbers: [The Common European Framework of Reference for Languages](https://www.languagecert.org/en/guided-learning-hours) has tabulate the guided learning hours typically required to reach each language proficiency level. These are hours spent studying with a teacher or structured learning. 

So for example, if you start as a beginner (A1) and want to become intermediate (B1) you will need approximately **400 guided hours**.  If you can dedicate **5 hours a week**, it will take **1.5 years**  and a minimum budget of **USD $6,000** to learn English (to give you an idea, for a person with minimum gauge in Argentina this cost is iqual to two year of salary). 

## Why can’t we rely on automatic translation only?

Why can’t one publish automatic translations directly, or rely on tools for on-the-fly translations?

> If you do that, the content in translated languages will be of a poorer quality so it’d mean accepting that users in some semi-supported language deserve material of less quality.

Reviewer: ```suggestion
It is a known fact that machine translation does not work so well for languages with smaller populations and
smaller online repositories.[^swidan]  Even for wel know languages, authomatic translation still have errors (like word meaning changes due to context, grammar and syntaxis aspect, and gender marks).   Machinne translation can't localize some content, like provide meaningful examples for the audience, idioms, currency and numerals symbols).  

If we rely on automatic translation only , the content in translated languages will be of a poorer quality so it’d mean accepting that users in some semi-supported language deserve material of less quality.
```

---
title: "Multilingual Publishing: Frequently Asked Questions"
author: 
- Yanina Bellini Saibene
- Maëlle Salmon
editor: 
date: '2025-06-13'
slug: translation-faq
output: hugodown::md_document
tags:
  - multilingual

>   - community

Author: @yabellini do you want us to add it?

When sharing about this project at [conferences](/talks-papers/) and [elsewhere](/commcalls/nov2023-multilingual/), we start an important conversation with people less used to such endeavours or to our particular [processes](https://translationguide.ropensci.org/).
In this post, we will summarize frequently asked questions, and our best answers to them (even better formulated than in the heat of a post-talk Q&A session :grin:).

## Will translations make people “lazy” in that they won’t learn English?

First of all, not learning English doesn’t mean being “lazy”. 
Working on one’s project with the tools at one’s disposal is already work enough. 
Besides, many people who do not speak English speak several languages.
Now, people might still choose to learn English later, for instance to take part in some conversations in English, but it won’t have been an issue for them to have first learnt how to, say, develop software using French.
Do we want people to learn how to use some tools or do we want them to be English speakers? 
In our case, we want people to be able to use and contribute to open-source tools without first having to become English speakers,

> because we are aware of the "manifold costs of being a non-native English speaker in science"[^amano].

Reviewer 2: You could also reiterate the idea of costs again in the conclusion? I agree that it's important, I just wasn't convinced that here was the best place (but happy to be overruled! 😉)

[^amano]: Amano T, Ramírez-Castañeda V, Berdejo-Espinola V, Borokini I, Chowdhury S, Golivets M, et al. (2023) The manifold costs of being a non-native English speaker in science. PLoS Biol 21(7): e3002184. https://doi.org/10.1371/journal.pbio.3002184

### Related: Is it that hard to read in English?

Yes, especially as it adds to the "manifold costs of being a non-native English speaker in science"[^amano] (we really like this paper :wink:).
There are several studies around the topic of learning to code in English base programming languages when you are not a English speaker. 

Gou and other[^gou] found that non-native English speakers faced barriers with reading instructional materials, technical communication, reading and writing code, and simultaneously learning English and programming. 
These difficulties are rooted in the need to develop English understanding at the same time that they are expected to learn about the specific topic, in our case how to develop research open source high quality software.

> 

Author: @yabellini ok with you?

> 

Reviewer 2: For me it's just my experience, I'm not sure of the literature.

Other results[^tshukudu] suggest that learning to program in English alone, when you are not a English speaker, isn't necessarily difficult for everyone, but including students' native language can make the learning experience clearer, more supportive and more enjoyable. 

> 

Author: @yabellini

## Why can’t we rely on automatic translation only?

Why can’t one publish automatic translations directly, or rely on tools for on-the-fly translations?
It is a known fact that machine translation does not work so well for languages with smaller populations and
smaller online repositories.[^swidan]  Even for wel know languages, authomatic translation still have errors (like word meaning changes due to context, grammar and syntaxis aspect, and gender marks).   Machinne translation can't localize some content, like provide meaningful examples for the audience, idioms, currency and numerals symbols).  

If we rely on automatic translation only , the content in translated languages will be of a poorer quality so it’d mean accepting that users in some semi-supported language deserve material of less quality.
This is a decision made at the organization level.

## Why don’t you use `<insert-name-of-translation-patform>` instead of Git and GitHub?

> We are a technical community translating technical content. 

Reviewer: ```suggestion
We are a technical community translating and localizing technical content. 
```

---
title: "Multilingual Publishing: Frequently Asked Questions"
author: 
- Yanina Bellini Saibene
- Maëlle Salmon
editor: Steffi LaZerte
date: '2025-06-13'
slug: translation-faq
output: hugodown::md_document
tags:
  - multilingual

>   - community

Reviewer: ```suggestion
  - community
  - diversity
```

---
title: "Multilingual Publishing: Frequently Asked Questions"
author: 
- Yanina Bellini Saibene
- Maëlle Salmon
editor: Steffi LaZerte
date: '2025-06-13'
slug: translation-faq
output: hugodown::md_document
tags:
  - multilingual
  - community
  - diversity

>   description: A list of frequently asked questions about our translation and localization process, and our best answers to them.

Reviewer: ```suggestion
  description: A list of frequently asked questions about our multilingual program and our translation and localization process, and our best answers to them.
```
